spec/lib/open_food_network/products_and_inventory_report_spec.rb

```diff
@@ -134,6 +134,32 @@ module OpenFoodNetwork
           subject.stub(:params).and_return(distributor_id: distributor.id)
           subject.filter(variants).should == [product2.variants.first]
         end
+
+        it "ignores variant overrides without filter" do
+          distributor = create(:distributor_enterprise)
+          product = create(:simple_product, supplier: supplier, price: 5)
+          variant = product.variants.first
+          order_cycle = create(:simple_order_cycle, suppliers: [supplier], distributors: [distributor], variants: [product.variants.first])
+          create(:variant_override, hub: distributor, variant: variant, price: 2)
+
+          result = subject.filter(variants)
+
+          expect(result.first.price).to eq 5
+        end
+
+        it "considers variant overrides with distributor" do
+          distributor = create(:distributor_enterprise)
+          product = create(:simple_product, supplier: supplier, price: 5)
+          variant = product.variants.first
+          order_cycle = create(:simple_order_cycle, suppliers: [supplier], distributors: [distributor], variants: [product.variants.first])
+          create(:variant_override, hub: distributor, variant: variant, price: 2)
+
+          allow(subject).to receive(:params).and_return(distributor_id: distributor.id)
```

Comment:
It is a bad practice to stub the subject of the test because we are not testing the actual implementation. Do you see a way to overcome this? I guess we need to pass a different set of params instead but I haven't checked the report's implementation.

Reply:
I agree that it's bad practice. But I think fixing it is beyond this pull request.

1. I used the same pattern as in the rest of the spec file. Solving this problem would be an improvement but it should be done for the whole file.
2. The `params` method is not provided by this controller but by one of the ancestors. Rails provides it. That's why I think it's okay to stub it. Ideally, all that filtering would happen in a service that is taking the params as input and we didn't need to stub it.

@sauloperez Are you okay to leave this as is? I don't like to get into refactoring work which then creates merge conflicts. Let's be stricter once this Spree upgrade is done.

Reply:
Fine, let's just create a tech debt issue then.